src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -1926,36 +1923,36 @@ void InitAirLoops(EnergyPlusData &state, bool const FirstHVACIteration) // TRUE
                     if (compType == CompType::Fan_Simple_CV || compType == CompType::Fan_Simple_VAV || compType == CompType::Fan_ComponentModel) {
                         if (thisPrimaryAirSys.OASysExists && !thisPrimaryAirSys.isAllOA) {
                             if (FoundOASys) {
-                                if (thisPrimaryAirSys.Branch(BranchNum).DuctType != DataHVACGlobals::AirDuctType::Heating) {
+                                if (thisPrimaryAirSys.Branch(BranchNum).DuctType != HVAC::AirDuctType::Heating) {
                                     SupFanIndex = Fans::GetFanIndex(state, thisPrimaryAirSys.Branch(BranchNum).Comp(CompNum).Name);
-                                    supFanModelType = StructArrayLegacyFanModels;
+                                    supFanType = state.dataFans->fans(SupFanIndex)->type;
```

Comment:
If CompIndex is set = to SupFanIndex here in InitAirLoops (called before SimAirLoopComponent) then there is no need to check for CompIndex == 0 in SimAirLoopComponent because it can never be 0 since it gets set here. 

    thisPrimaryAirSys.Branch(BranchNum).Comp(CompNum).CompIndex = SupFanIndex;

            SimAirLoopComponent(state,
                                PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).Name,
                                CompType_Num,
                                FirstHVACIteration,
                                AirLoopNum,
                                PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).CompIndex,
                                PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).compPointer,
                                AirLoopNum,
                                BranchNum,
                                CompNum);

I just saw that this is now also happening in GetAirPathData but only for the SystemFan model. The end result is the same and this could happen in either place. It's probably better to do all this setup processing in GetAirPathData but that does not have to happen in this branch. One way or another these fan (or all comps for that matter) indexs need to be set early so that there is no need to check if CompIndex == 0 where these comps are called to simulate.

                    } else if (componentType == "FAN:SYSTEMMODEL") {
                        primaryAirSystems.Branch(BranchNum).Comp(CompNum).CompType_Num = CompType::Fan_System_Object;
                        auto &comp = primaryAirSystems.Branch(BranchNum).Comp(CompNum);
                        if (comp.CompIndex == 0) {
                                comp.CompIndex = Fans::GetFanIndex(state, comp.Name); // TODO: get rid of this
                            if (comp.CompIndex == 0) {
                                ShowSevereError(state, format("Component {} of type {} not found.", comp.Name, comp.TypeOf));
                            }
                        }
                                
                        state.dataFans->fans(comp.CompIndex)->airPathFlag = true;
                    } else if (componentType == "FAN:COMPONENTMODEL") {